### DISCUSSION TITLE
VMDK Image Mounting Script

### DISCUSSION THREAD

**Discussion by RichardBachelier** [ltsp/ltsp#883] [General] [answered]
Hello,

I've created a script to perform various mounts and binds to 'chroot' into my VMDK in order to regularly update my image (update, install, etc.). Could you please let me know if the instructions in my script are sufficient or not?

Thank you for your help.

**Comment by RichardBachelier** [comment]
```bash
#/bin/sh

MNTIMAGE='/srv/mnt-image'

if [ $# -ne 2 ]; then
  echo "Ce script requiere 2 paramétres :"
  echo " - Nom de l'image vmdk"
  echo " - chroot oui / non "
  exit
fi

modprobe nbd
qemu-nbd -c /dev/nbd1 /srv/ltsp/images/$1
sleep 2
mount /dev/nbd1p1 $MNTIMAGE
mount --bind /proc $MNTIMAGE/proc
mount --bind /etc/resolv.conf $MNTIMAGE/etc/resolv.conf
mount -t devpts none $MNTIMAGE/dev/pts
mount --bind  /dev $MNTIMAGE/dev
mount --bind  /run $MNTIMAGE/run
mount --bind /var/run/dbus/ $MNTIMAGE/var/run/dbus/
mount -t devpts none $MNTIMAGE/dev/pts


if [ $2 == "oui" ]
then
  chroot $MNTIMAGE
fi
```

**Comment by alkisg** [comment, marked as answer]
Just run `systemd-nspawn -i /path/to/vm-flat.vmdk`.

**Comment by RichardBachelier** [comment]
Thank you very much for your prompt response. 

I have one last technical question about the LTSP processes. My VMDK file is 35GB. When I generate the IMG file using the 'ltsp image' command, it takes some time (at least 30 minutes). This is probably normal as we are compressing the VMDK. Is there an option in this command to speed up this compilation, considering that I have 16GB of RAM and 4 CPUs?

**Comment by alkisg** [comment]
How big is the resulting, compressed image?
What's the output of `sudo hdparm -t /dev/sda` (or whatever you disk is)?
That command will show your disk speed. What matters in `ltsp image` is the CPU speed and the disk speed, if it's HDD or SSD.

**Comment by RichardBachelier** [comment]
/dev/sda: Timing buffered disk reads: 340 MB in  3.06 seconds = 111.29 MB/sec
compresses image size about 11 G0
SSD disk,  Intel(R) Xeon(R) CPU E5-2609 0 @ 2.40GHz 

I think that this is normal, then.

**Comment by alkisg** [comment]
11 GB is too big, a normal one is 2-5 GB.
Are you sure you don't have any global /data directories that should have been excluded from the target image?

HDD read speeds are around 200 MB/sec
SSD read speeds are around 500 MB/sec
NVMe read speeds are more than 2000 MB/sec

Your disks is extremely slow for a server (well, unless it was in use by the users while you were running that command).

**Comment by RichardBachelier** [comment]
In my VMDK, I have 25GB of data, which may explain the size of the compressed file
We will also check the performance of our hard drives.

thank you

**Comment by alkisg** [comment]
Data usually goes on the server, not in the image.
